### PR TITLE
changes to add paralus addon to eks blueprint

### PR DESCRIPTION
**Pull request by niravparikh05**
*Issue #, if available:*

https://github.com/aws-quickstart/cdk-eks-blueprints/issues/580

*Description of changes:*

Creates a sample eks blueprint for paralus addon

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by niravparikh05** [comment]
@parkand1 Lately I am running into below issue while running any cdk commands, even in main branch. Any idea on what's causing this and how do I fix it ?

Node Version: v16.19.1
npm version: 8.19.3
cdk version: 2.65.0

/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/sync-rpc/lib/index.js:165
    const error = new Error(msg.v.message);
                  ^
Error: read ECONNRESET
    at extractValue (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/sync-rpc/lib/index.js:165:19)
    at /home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/sync-rpc/lib/index.js:175:12
    at request (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/sync-request/lib/index.js:28:15)
    at loadExternalYaml (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/@aws-quickstart/eks-blueprints/lib/utils/yaml-utils.ts:46:22)
    at CsiDriverProviderAws.deploy (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/@aws-quickstart/eks-blueprints/lib/addons/secrets-store/csi-driver-provider-aws.ts:54:61)
    at SecretsStoreAddOn.deploy (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/@aws-quickstart/eks-blueprints/lib/addons/secrets-store/index.ts:66:53)
    at new EksBlueprint (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/node_modules/@aws-quickstart/eks-blueprints/lib/stacks/eks-blueprint-stack.ts:263:34)
    at new MultiTeamConstruct (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/lib/multi-team-construct/index.ts:42:9)
    at Object.<anonymous> (/home/infracloud/development/vscode-workspace/aws-samples/cdk-eks-blueprints-patterns/bin/main.ts:35:1)
    at Module._compile (node:internal/modules/cjs/loader:1191:14) {
  code: 'ECONNRESET'
}

**Comment by parkand1** [comment]
@niravparikh05 I just tried a new deploy from the patterns repo and it is also working on my end

```bash
blueprints-addon-paralus-contour-contour-6bd75d995f-zrf9n   1/1     Running   0             64s
blueprints-addon-paralus-contour-envoy-dj5n8                2/2     Running   0             64s
blueprints-addon-paralus-fluent-bit-w7tnf                   1/1     Running   2 (48s ago)   64s
blueprints-addon-paralus-kratos-7cc678d95b-g2l9d            2/2     Running   0             64s
blueprints-addon-paralus-kratos-courier-0                   2/2     Running   0             63s
blueprints-addon-paralus-postgresql-0                       1/1     Running   0             63s
dashboard-79f8d5d6d-8lt7b                                   1/1     Running   0             64s
paralus-856ccbf875-5pn7s                                    2/2     Running   0             64s
prompt-7677588bcc-vjqhh                                     2/2     Running   0             64s
relay-server-6cb9f6cf86-d5jw9                               2/2     Running   1 (23s ago)   63s
```

The only thing I'd like to add is an example of a sample use-case in the documentation and from there I am good to merge!

**Comment by niravparikh05** [comment]
thanks @parkand1 I have added some information around a sample use-case with reference to a detailed blog around it.

**Comment by niravparikh05** [comment]
@parkand1 just following up to check if you got a chance to review the documentation changes ?

File: lib/paralus-construct/index.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                    // this signifies the paralus version to be deployed
+                    version: '0.2.2',
+                    // this flag deploys kratos in dev mode, install postgres, by default it is true
+                    development: true,
+                    /**
+                    * Values to pass to the chart as per https://github.com/paralus/helm-charts/blob/main/charts/ztka/values.yaml.
+                    */
+                    // update this to your domain, as paralus works based on domain based routing
+                    values: {
+                       "fqdn.domain": "paralus.local"
```
Comment (parkand1) [inline lib/paralus-construct/index.ts:29]: Can we update this based on our conversation today?

```bash
                    values: {
                        "fqdn": {
                         "domain": "your.domain",
                         "hostname": "console-ic-oss",
                         "coreConnectorSubdomain": "*.core-connector.ic-oss",
                         "userSubdomain": "*.user.ic-oss"
                        }
```

**Review by parkand1 — CHANGES_REQUESTED** [review]

**Review by parkand1 — CHANGES_REQUESTED** [review]
Added a couple more comments

File: docs/patterns/paralus.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Paralus on EKS
+The Paralus project is a free, open source tool that enables controlled, audited access to Kubernetes infrastructure. It comes with just-in-time service account creation and user-level credential management that integrates with your RBAC and SSO. 
+
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply docs/patterns/paralus.md:3]: Added a learn more link to paralus website.

File: docs/patterns/paralus.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+This pattern deploys the following resources:
+
+- Creates EKS Cluster Control plane with public endpoint (for demo purpose only) with a managed node group
+- Deploys supporting add-ons:  AwsLoadBalancerController, VpcCni, KubeProxy, EbsCsiDriverAddOn
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply docs/patterns/paralus.md:7]: Added a note on the add-on dependencies

File: docs/patterns/paralus.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+blueprints-addon-paralus-contour-contour-7857f4cd9-kqhgp   1/1     Running                 
+blueprints-addon-paralus-contour-envoy-mx8z7               2/2     Running                 
+blueprints-addon-paralus-fluent-bit-525tt                  1/1     Running                 
+blueprints-addon-paralus-kratos-588775bc47-wf5gf           2/2     Running                 
+blueprints-addon-paralus-kratos-courier-0                  2/2     Running                 
+blueprints-addon-paralus-postgresql-0                      1/1     Running                 
+dashboard-6d8b54d78b-d8cks                                 1/1     Running                 
+paralus-66d9bbf698-qznzl                                   2/2     Running                 
+prompt-54d45cff79-h9x95                                    2/2     Running
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply docs/patterns/paralus.md:79]: probably paralus pod took a tad bit longer due to its dependencies on kratos, postgres etc., prompt attempts to verify dependent paralus services and restarts if paralus is not ready yet. 

similar behaviour is seen for relay, however it should be in running once paralus is up.

File: docs/patterns/paralus.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+Let’s verify the resources created by Steps above.
+```sh
+kubectl get nodes # Output shows the EKS Managed Node group nodes
+
+kubectl get ns | grep paralus # Output shows paralus namespace
+
+kubectl get pods --namespace=paralus-system  # Output shows paralus pods
+
+blueprints-addon-paralus-contour-contour-7857f4cd9-kqhgp   1/1     Running
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply docs/patterns/paralus.md:71]: added a link detailing all the various components used in paralus.

File: lib/paralus-construct/index.ts (hunk 1 above)
Comment (niravparikh05) [reply lib/paralus-construct/index.ts]: updated accordingly.

**Comment by niravparikh05** [comment]
@parkand1 addressed all the comments and updated the pr. kindly have a look at your convenience.

**Comment by niravparikh05** [comment]
@parkand1 just following up to check if you got a chance to review the changes ?

**Comment by niravparikh05** [comment]
@parkand1 following up on this, did you get a chance to try this out ? Looking to take this forward.

**Review by parkand1 — APPROVED** [review]
LGTM

**Pull request merged**